### DISCUSSION TITLE
Backup of config and Trusted Proxy

### DISCUSSION THREAD

**Discussion by JuliDi** [Part-DB/Part-DB-server#313] [Q&A] [answered]
Hi,
[the documentation says](https://docs.part-db.de/usage/backup_restore.html#backup-manual) that for manual backup, I should copy the `.env.local` (cant find that, there is `.env.docker` however) and `config/parameters.yaml` and `config/banner.md`
The question is: Why are these files important? The config folder is not bound to a volume in docker compose as [described here](https://docs.part-db.de/installation/installation_docker.html) so they are not persistent anyway:
```yaml
      - ./uploads:/var/www/html/uploads
      - ./public_media:/var/www/html/public/media
      - ./db:/var/www/html/var/db
```

What am I missing here? And what would a correct backup look like with docker compose?

And one more question: What is the correct TRUSTED_PROXIES IP for my webserver that is running outside of docker? Is it just the server's IP or localhost?

Thanks

**Comment by jbtronics** [comment, marked as answer]
They are only important to backup, if you have done changes to it (in that case you would map them to a volume, if you use docker).
That should not be needed in most cases, but you can change some less important configuration stuff in the  `parameters.yaml` file. And for non docker installations it might be easier to change the content of the banner.md instead of the .env file

For docker installations its normal that there is no `.env.local` file, as the environment files are defined via docker itself. In that case you have to save the configuration for the docker container (e.h. the `docker-compose.yaml` file in case of using docker-compose)

**Reply by JuliDi** [reply]
Cool, thanks for the explanation. Then I will just save my docker-compose file in addition.

**Reply by JuliDi** [reply]
Do I need to set the TRUSTED_PROXIES with my setup or is that not necessary then?

**Reply by jbtronics** [reply]
Depends on if you have a reverse proxy before your docker container (in that case you would need to set it to the IP address of the reverse proxy). If you access Part-DB directly, then thats not needed.

The TRUSTED_PROXIES option is only important to get the correct IP addresses of end users. If you use an reverse proxy and has not configured this setting, then the wrong IP address will get logged to the server log (the IP of the reverse proxy instead of the one of the user). Thats basically the only thing for which this is important.

**Reply by JuliDi** [reply]
I have the reverse proxy running on the server directly, not in docker. So it should work fine if I just enter 127.0.0.1, right?

**Reply by jbtronics** [reply]
No you need to put the IP address Part-DB "sees" from your reverse proxy. That depends a bit on your docker configuration, but in default that should be: 172.17.0.1
